state/watcher/watcher_test.go

```diff
@@ -703,3 +705,72 @@ func (s *SlowPeriodSuite) TestStartSyncStartsImmediately(c *gc.C) {
 	case <-time.After(justLongEnough):
 	}
 }
+
+type badIter struct {
+	*mgo.Iter
+
+	errorAfter *int
```

Comment:
Why a pointer? Since the interface is supported by a *badIter, you can just use a normal int here.